internal/pkg/service/stream/storage/node/coordinator/filerotation/operator.go

```diff
@@ -336,35 +366,40 @@ func (o *operator) closeFile(ctx context.Context, file *fileData) {
 	// If there is an error, increment retry delay
 	if err != nil {
 		o.logger.Error(ctx, err.Error())
-		err = o.storage.File().IncrementRetryAttempt(file.FileKey, o.clock.Now(), err.Error()).Do(ctx).Err()
+		err = o.storage.File().IncrementRetryAttempt(file.FileKey, o.clock.Now(), err.Error()).RequireLock(lock).Do(ctx).Err()
 		if err != nil {
-			o.logger.Errorf(ctx, "cannot increment file retry", err)
+			o.logger.Errorf(ctx, "cannot increment file close retry", err)
+			return
 		}
-		return
 	}
 
 	// Prevents other processing, if the entity has been modified.
 	// It takes a while to watch stream send the updated state back.
 	file.Processed = true
+
+	if err == nil {
+		o.logger.Info(ctx, "successfully closed file")
+	}
 }
 
-func (o *operator) waitForSlicesUpload(ctx context.Context, fileKey model.FileKey) error {
+func (o *operator) waitForSlicesUpload(ctx context.Context, fileKey model.FileKey) (allEmpty bool, err error) {
 	for {
 		// Order is important, to make it bulletproof, we have to get the notifier channel before the check
```

Comment:
:thinking: 

Maybe `allEmpty` shouldn't be part of this method.

It can have own lock and own method, or be inlined to the `closeFile`.

It is not directly related to waiting.